### ISSUE TITLE
Enhance code for keyboard handling

### ISSUE THREAD

**Issue by s-macke** [s-macke/jor1k#26]
Correct handling of the input is a very difficult task in Javascript. 
At the moment, jor1k supports two input devices. 
1. A terminal input
2. A opencore-keyboard device. 

Between 1. and 2. is switched, if the framebuffer or the terminal is selected.
The code to handle both keyboards is inconsistent and merged together in form of a small hack. In order to allow multiple terminals in future, it is necessary to clean up the whole keyboard input code.

**Comment by neelabhg** [comment]
I added the [Terminal Input/Output Wiki Page](https://github.com/s-macke/jor1k/wiki/Terminal-Input-Output) to get help on this issue.

**Comment by s-macke** [comment]
Yes, that's basically how it works for the terminal and UART. Nice summary. Still I don't like the keyboard handling.
